lib/server.js

```diff
@@ -81,6 +81,29 @@ exports.create = function create(createOptions) {
       callback(err, html);
     }
 
+    function matching(options) {
+      var componentInstance = null;
+      match(options, function(error, redirectLocation, renderProps) {
+        if (redirectLocation) {
+          debug('server.js match 302 %s', JSON.stringify(redirectLocation));
```

Comment:
@chunkiat82 there is a potential chance that this can go in an infinite loop. Also, a 302 should really hit the browser and then get redirected